### DISCUSSION TITLE
Installing Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by ahmedelq** [microsoft/WSL#11515] [Q&A]
Hi everyone!

Would love to upgrade my Ubuntu to 24.04 since it has been released. However,  ` wsl --list --online` gives the below:

```
Ubuntu                                 Ubuntu
Debian                                 Debian GNU/Linux
kali-linux                             Kali Linux Rolling
Ubuntu-18.04                           Ubuntu 18.04 LTS
Ubuntu-20.04                           Ubuntu 20.04 LTS
Ubuntu-22.04                           Ubuntu 22.04 LTS
OracleLinux_7_9                        Oracle Linux 7.9
OracleLinux_8_7                        Oracle Linux 8.7
OracleLinux_9_1                        Oracle Linux 9.1
openSUSE-Leap-15.5                     openSUSE Leap 15.5
SUSE-Linux-Enterprise-Server-15-SP4    SUSE Linux Enterprise Server 15 SP4
SUSE-Linux-Enterprise-15-SP5           SUSE Linux Enterprise 15 SP5
openSUSE-Tumbleweed                    openSUSE Tumbleweed
``` 

Will it be included? How can we install it? 
Thanks

**Comment by elsaco** [comment]
@ahmedelq grab the latest release from https://cloud-images.ubuntu.com/wsl/noble/ or wait untill 24.04 makes it into the store.

**Comment by Dranduck** [comment]
... and it's in the store
`PS C:\Windows\system32> wsl --list -o
The following is a list of valid distributions that can be installed.
Install using 'wsl.exe --install <Distro>'.

NAME                                   FRIENDLY NAME
Ubuntu                                 Ubuntu
Debian                                 Debian GNU/Linux
kali-linux                             Kali Linux Rolling
Ubuntu-18.04                           Ubuntu 18.04 LTS
Ubuntu-20.04                           Ubuntu 20.04 LTS
Ubuntu-22.04                           Ubuntu 22.04 LTS
**Ubuntu-24.04                           Ubuntu 24.04 LTS**
OracleLinux_7_9                        Oracle Linux 7.9
OracleLinux_8_7                        Oracle Linux 8.7
OracleLinux_9_1                        Oracle Linux 9.1
openSUSE-Leap-15.5                     openSUSE Leap 15.5
SUSE-Linux-Enterprise-Server-15-SP4    SUSE Linux Enterprise Server 15 SP4
SUSE-Linux-Enterprise-15-SP5           SUSE Linux Enterprise 15 SP5
openSUSE-Tumbleweed                    openSUSE Tumbleweed`

**Comment by ethernic** [comment]
I wonder if this is related to the note [here](https://ubuntu.com/tutorials/upgrading-ubuntu-desktop#1-before-you-start) (for 22.04 LTS) that the upgrade wouldn't appear until this first point release. I installed the current version separately and can confirm it does not have a point release (it's just 24.04 LTS, compared to my current 22.04.4 LTS.)

If that's the case, it seems your options are to install fresh or wait until the first point release to do the upgrade.

**Reply by pratnala** [reply]
I am on 23.10 and still don't see the upgrade

**Reply by mickleby** [reply]
Apparently the upgrade is waiting on some "blocker bugs" to be resolved: https://discourse.ubuntu.com/t/noble-numbat-24-04-release-status-tracking/44043

**Reply by wolf99** [reply]
Looks like 24.04.01 is still not released quite yet: https://discourse.ubuntu.com/t/noble-numbat-24-04-1-lts-point-release-status-tracking/46972 
There are some upgrade blockers and release blockerss
Current plan there is to release on 2024-08-29.

**Reply by wolf99** [reply]
First point version  24.04.01 is now released. If the above logic from @ethernic holds true, then we should expect to see the update on the store soon.

**Reply by ImpatientTurtle** [reply]
Just upgraded using do-release-upgrade
![WindowsTerminal_1725210332](https://github.com/user-attachments/assets/2e0887e2-3185-4aa8-9c5b-5925a9b96bcd)

**Reply by Sophist-UK** [reply]
```
sophist@PC:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```
```
sophist@PC:~$ sudo do-release-upgrade
Checking for a new Ubuntu release
There is no development version of an LTS available.
To upgrade to the latest non-LTS development release
set Prompt=normal in /etc/update-manager/release-upgrades.
```
```
sophist@PC:~$ sudo do-release-upgrade -c -d
Checking for a new Ubuntu release
New release '24.04.1 LTS' available.
Run 'do-release-upgrade' to upgrade to it.
```

Running `sudo do-release-upgrade -d` gives errors.
